### ISSUE TITLE
Retrieving orders in sectors with WaitDeals status caused miner's fil-markets to get stuck

### ISSUE THREAD

**Issue by IPFSMain-Official** [filecoin-project/lotus#3187]
**Describe the bug**
1. This data is still in WaitDeals state in miner's corresponding sector, and then I accidentally execute Retrieve on the client side, and it's going to stay stuck here:  
```shell
$ lotus client retrieve bafykbzacecemdfcyngard4c4hykeni44xbjtbdckn2pvwryncc2ma3d2a6jog /home/ipfsmain/lotus-nor/lotus-from-miner.zip
> Recv: 0 B, Paid 0 FIL, ClientEventOpen (DealStatusNew)
> Recv: 0 B, Paid 0 FIL, ClientEventDealProposed (DealStatusWaitForAcceptance)
> Recv: 0 B, Paid 0 FIL, ClientEventDealAccepted (DealStatusAccepted)
> Recv: 0 B, Paid 0 FIL, ClientEventPaymentChannelAddingFunds (DealStatusPaymentChannelAddingFunds)
> Recv: 0 B, Paid 0 FIL, ClientEventPaymentChannelReady (DealStatusOngoing)
^CERROR: retrieval timed out
```

2. At this point, the corresponding retrieve state in miner has already failed:  
```shell
$ lotus-miner retrieval-deals list
12D3KooWMHf7UeG8Dd82qc6P7hHTETE7qzcXY2H4SvbsKx53gF2m  2       ...3d2a6jog  DealStatusFailed     2             0          draining piece reader: copy sealed/cache sector data: failed to acquire sector {2098 14} from remote(2): sector not found
```

3. So once again the retrieve would be stuck here:  
```shell
$ lotus client retrieve --miner t02098 bafk2bzacecb42hp6sgqm3rly7ksr75bdgiur7et4q4qj7bm6mxv6mmy537xzu /home/ipfsmain/lotus-nor/txjson-from-miner
> Recv: 0 B, Paid 0 FIL, ClientEventOpen (DealStatusNew)
> Recv: 0 B, Paid 0 FIL, ClientEventDealProposed (DealStatusWaitForAcceptance)
^CERROR: retrieval timed out
```

4. Once again:  
```shell
$ lotus client retrieve --miner t02098 bafk2bzacecb42hp6sgqm3rly7ksr75bdgiur7et4q4qj7bm6mxv6mmy537xzu /home/ipfsmain/lotus-nor/txjson-from-miner
Success
```

5. Restart client side lotus, once again still stuck here:  
```shell
$ sudo systemctl restart lotus-daemon-nor.service
$ lotus client retrieve --miner t02098 bafk2bzacecb42hp6sgqm3rly7ksr75bdgiur7et4q4qj7bm6mxv6mmy537xzu /home/ipfsmain/lotus-nor/txjson-from-miner
> Recv: 0 B, Paid 0 FIL, ClientEventOpen (DealStatusNew)
> Recv: 0 B, Paid 0 FIL, ClientEventDealProposed (DealStatusWaitForAcceptance)
^CERROR: retrieval timed out
```

6. If a storage order is initiated at this point, the storage order is stuck at StorageDealTransferring:
```shell
$ lotus client list-deals
bafyreictc2l422leqlcfbpdkhtkgm3swcn5sdvgamnteuq5d4maa6bg2mq  0       t02098    StorageDealTransferring   N          N         baga6ea4seaqni5a7ybviic2hoaatkrgkvcudo65lsevj3uecq3qx2wf6qh5didq  1.984 MiB  0.0002602935 FIL  520587 
```

It was only after miner had been restarted that all storage and retrieval became normal, but miner would not work properly again if the WaitDeals data was retrieved again.

**Version (run `lotus version`):**
lotus version 0.4.6+git.5c6b99c9

**Comment by Reiers** [comment]
Hi @IPFSMain-Official 

Thanks for the report. Seems like the issue is fixed. 
Please upgrade [Lotus.](https://github.com/filecoin-project/lotus/releases/tag/v1.15.1)

Closing Lotus Issue ticket.

If you are still experiencing problems, open a new ticket or ask for help in our [Discussions forum.](https://github.com/filecoin-project/lotus/discussions)
